### PR TITLE
Mes 8240 eco fields bug fix

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Fixed bugs that stop fuel efficiency card from functioning properly 
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image

### CODE REVIEW

File: src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html [hunk 1]
```diff
@@ -1,4 +1,5 @@
-<ion-row [formGroup]="formGroup" class="mes-validated-row" *ngIf="fuelEfficientDriving">
+<ion-row class="mes-row-separator" *ngIf="fuelEfficientDriving"></ion-row>
+    <ion-row [formGroup]="formGroup" class="mes-validated-row mes-row-separator" *ngIf="fuelEfficientDriving">
```
Comment (matthew2564) [inline src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html]: Does ```*ngIf``` need to be on both?

File: src/app/pages/office/cat-adi-part2/office.cat-adi-part2.page.ts [hunk 2]
```diff
@@ -217,6 +217,17 @@ export class OfficeCatADI2Page extends OfficeBasePageComponent implements OnInit
...
+    let display: boolean;
+    combineLatest([
+      this.pageState.seriousFaults$,
+      this.pageState.dangerousFaults$,
+      this.pageState.adi2DrivingFaults$]).subscribe((data) => {
+      display = data[0].length === 0 && data[1].length === 0 && data[2].length === 0;
+    });
+    return display;
+  }
+
```
Comment (matthew2564) [inline src/app/pages/office/cat-adi-part2/office.cat-adi-part2.page.ts]: Lets try and move this to the pageState object and reference this in the HTML using async pipe.

Comment (matthew2564) [reply src/app/pages/office/cat-adi-part2/office.cat-adi-part2.page.ts]: i.e. Inside page state, create a new property called 

```
  displayFuelEfficient$: Observable<boolean>;
```

We could try something like this
```
      displayFuelEfficient$: currentTest$.pipe(
        withLatestFrom(
          this.pageState.seriousFaults$,
          this.pageState.dangerousFaults$,
          this.pageState.adi2DrivingFaults$,
        ),
        map((
          [, seriousF, dangerousF, drivingF],
        ) => !!(seriousF?.length === 0 && dangerousF?.length === 0 && drivingF?.length === 0)),
      ),
```

And then in the template

```
pageState.displayFuelEfficient | async
```

Comment (RLCorp) [reply src/app/pages/office/cat-adi-part2/office.cat-adi-part2.page.ts]: addressed in newest commit

File: src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html (hunk 1 above)
Comment (RLCorp) [reply src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html]: addressed in newest commit

**Review by matthew2564 — APPROVED** [review]

**Review by gittins7 — APPROVED** [review]

**Pull request merged**